specs/0118-notification-center.md

```diff
@@ -0,0 +1,119 @@
+    - Dropdown list: `Activity type`
+      - Options: `Select activity type`, `Submitted translations`, `Performed reviews`, `Last login`
+      - Select one
+      - Default value: `Select activity type`
+      - Form dynamically displays the appropriate filter inputs below based on the selection in the dropdown. `Select activity type` does not display any filter fields.
+      - Validation:
+        - Multiple activity filters can used at the same time
+        - Error if more than one filter of the same type used: `You cannot use multiple of the same activity type.`
+        - Error if `Select activity type` selected as a filter in the dropdown: `You must select an activity type when you add an activity filter.`
+    - Filter inputs:
+      - Inputs for `Submitted translations` / `Performed reviews`
+        - `Activity level`
+          - Input box: `Minimum`  (value required, default value: 0, only accepts non-negative integers)
+          - Input box: `Maximum` (value optional, default value: empty, leaving empty will mean there is no maximum and all numbers greater than or equal to the value in `Minimum` will be included.
+          - Tooltip on `Maximum`: `Leave empty to include all numbers greater than or equal to Minimum.`
+        - `Timeframe filter`
+          - Checkbox (default empty): `Enable timeframe filter`
+            - Checking the box displays datepickers
+          - Datepicker: `From:` (Default value: today's date)
+          - Datepicker: `To:` (Default value: today's date)
+      - Inputs for `Last login`
+        - `Timeframe filter`
+          - Datepicker: `From:` (Default value: today's date)
+          - Datepicker: `To:` (Default value: today's date)
+- Header: `Transactional email`
+  - By default, emails are sent to users who have enabled opt-in consent. (Does not apply to in-app notifications.)
+  - This “Transactional” option flags emails as transactional and sends emails to users regardless of their opt-in status. However, this email content would be restricted to content transactional in nature (e.g. account status notifications, etc.).
+  - Checkbox (default empty): `This is a transactional email`
+  - Text description: `Transactional emails are sent to users who have not opted in to email communication. Transactional emails are restricted in the type of content that can be included.`
+  - Validation:
+    - Error if `Transactional email` enabled but `Message type` does not include `Email`: `You cannnot enable the transactional email option if the Message type does not include email.`
+- Button: `Review message`
+  - A send button is not displayed on the same page as the message editor and recipient selection to avoid sending an incomplete message or a message to the wrong recipients.
+  - When pressed a validation step is performed on inputs to ensure there are no issues with the form. If there are no issues, the user proceeds to the Review/Confirmation page.
+
+- Title: `Review message`
```

Comment:
I'm not sure there's a lot of value in having such Review step in addition to Preview.

Instead, having the ability to send the message to herself would let the user see the in-app notification or the email in the same environment the recipients will see it, revealing potential issues the Review step would hide.

Reply:
The review step isn't just for reviewing the message, it's also a sanity check to make sure that you are sending to the people you meant to and catch any other errors - and sending to yourself isn't going to check that. Also, from a lot of past experience I will catch errors during a review step so personally I see value in it. Especially since we can't take back sending mass emails with potential errors / or incorrect recipients - the chance of making a mistake is large and the potential for impact is large as well, so being extra cautious is warranted here.

Reply:
+1 on having an explicit review step on top of preview (note that it will display the number of recipients, which we probably don't want to calculate on the fly).

With that said, I like the idea of also having a button to send the message to yourself.

Reply:
I didn't phrase my initial point exactly as I wanted. Let's try again. :)

I agree the Review step is important, but I'm wondering if we need Review AND Preview (AND Ability to send a test message). I know in the ideal World we'd have all, but this spec if already quite big.

Sending a **test message** is IMO a must, because that's the only way to see how the email or the notification actually renders. I use it for every mass email I send from Pontoon and we make corrections in about 50% of cases.

**Review**, as Bryan mentions, also checks for errors outside the message content. I'd keep that, too.

**Preview** on the other hand is not as reliable as a test message, and is enforced by the Review step. Yes, it lets you check you message instantly, but that's not as critical as the other two. So I'd treat is as optional.

Reply:
So, to clarify, are you suggesting to remove the rendered view of the message in the Review step, and the "Preview" tab from the messaging editor?

Reply:
Just the "Preview" tab.